ocaml/xapi/daemon_manager.mli

```diff
@@ -0,0 +1,50 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+type daemon_check =
+	| Pidfile of string
+		(** Check whether the daemon is running by reading a pidfile, and checking
+		    that the PID points to a running process. *)
+	| Function of (unit -> bool)
+		(** Generic user-defined check, *)
+
+type daemon_state = [
```

Comment:
Actually, `daemon_state` doesn't need to be in the interface at all. I'll take it out.